hooks/tk-nuke_actions.py

```diff
@@ -183,15 +183,21 @@ def _create_read_node(self, path, sg_publish_data):
         if ext.lower() not in valid_extensions:
             raise Exception("Unsupported file extension for '%s'!" % path)
 
+        # This form will extract the format and frame range from the file
+        # itself. We can also see if there's a matching template and override
+        # the frame range, but this should handle the zero config case. This
+        # will also automatically extract the format and frame range for movie
+        # files.
+        read_node = nuke.createNode("Read")
```

Comment:
Hi Josh, the code looks alright, but I have no idea what is going on. To me it looks like you’re just instantiating a node and setting its values one by one instead of the old code which instantiated Node with all values already set.

Why the change?